e3sm_diags/driver/polar_driver.py

```diff
@@ -1,264 +1,400 @@
-            if ref_name == "WARREN":
-                # This is cdms2 fix for bad mask, Denis' fix should fix this.
-                mv2 = MV2.masked_where(mv2 == -0.9, mv2)
-            # The following should be moved to a derived variable.
-            if ref_name == "AIRS":
-                # This is cdms2 fix for bad mask, Denis' fix should fix this.
-                mv2 = MV2.masked_where(mv2 > 1e20, mv2)
-            if ref_name == "WILLMOTT" or ref_name == "CLOUDSAT":
-                # This is cdms2 fix for bad mask, Denis' fix should fix this.
-                mv2 = MV2.masked_where(mv2 == -999.0, mv2)
-
-                # The following should be moved to a derived variable.
-                if var == "PRECT_LAND":
-                    days_season = {
-                        "ANN": 365,
-                        "DJF": 90,
-                        "MAM": 92,
-                        "JJA": 92,
-                        "SON": 91,
-                    }
-                    # mv1 = mv1 * days_season[season] * 0.1 # following AMWG
-                    # Approximate way to convert to seasonal cumulative
-                    # precipitation, need to have solution in derived variable,
-                    # unit convert from mm/day to cm.
-                    mv2 = (
-                        mv2 / days_season[season] / 0.1
-                    )  # Convert cm to mm/day instead.
-                    mv2.units = "mm/day"
-
-            # For variables with a z-axis.
-            if mv1.getLevel() and mv2.getLevel():
-                plev = parameter.plevs
-                logger.info("Selected pressure level: {}".format(plev))
-
-                mv1_p = utils.general.convert_to_pressure_levels(
-                    mv1, plev, test_data, var, season
-                )
-                mv2_p = utils.general.convert_to_pressure_levels(
-                    mv2, plev, ref_data, var, season
-                )
+    if ds_ref is not None:
+        var_ref = ds_ref[var_key]
 
-                # Select plev.
-                for ilev in range(len(plev)):
-                    mv1 = mv1_p[ilev,]
-                    mv2 = mv2_p[ilev,]
-
-                    for region in regions:
-                        logger.info("Selected region: {}".format(region))
-
-                        mv1_domain = utils.general.select_region(
-                            region, mv1, land_frac, ocean_frac, parameter
-                        )
-                        mv2_domain = utils.general.select_region(
-                            region, mv2, land_frac, ocean_frac, parameter
-                        )
-
-                        parameter.output_file = "-".join(
-                            [
-                                ref_name,
-                                var,
-                                str(int(plev[ilev])),
-                                season,
-                                region,
-                            ]
-                        )
```

Comment:
Corresponds to:
`io.py: _save_data_metrics_and_plots`
```
    output_dir = _get_output_dir(parameter)
    filename = f"{parameter.output_file}.json"
    filepath = os.path.join(output_dir, filename)
```

Or perhaps:
`io.py: _write_vars_to_netcdf`:
```
    dir_path = _get_output_dir(parameter)
    filename = f"{parameter.output_file}_output.nc"
    output_file = os.path.join(dir_path, filename)
```

Probably the former though, since we already have the `utils.general.save_ncfiles -> _write_vars_to_netcdf` function correspondence below.

Reply:
Actually, this corresponds to:
`_run_diags_3d`:
```
parameter._set_param_output_attrs(var_key, season, region, ref_name, ilev)
```